src/experiments/index.js

```diff
@@ -125,13 +125,18 @@ export function experimentToggles(win) {
   win[TOGGLES_WINDOW_PROPERTY] = map();
   const toggles = win[TOGGLES_WINDOW_PROPERTY];
 
-  // Read the default config of this build.
-  if (win.AMP_CONFIG) {
-    for (const experimentId in win.AMP_CONFIG) {
-      const frequency = win.AMP_CONFIG[experimentId];
-      if (typeof frequency === 'number' && frequency >= 0 && frequency <= 1) {
-        toggles[experimentId] = Math.random() < frequency;
-      }
+  // eslint-disable-next-line local/window-property-name
+  win.AMP_EXP ??= JSON.parse(win.__AMP_EXP?.textContent || '{}');
+
+  // Read default and injected configs of this build.
+  const buildExperimentConfigs = {
+    ...(win.AMP_CONFIG ?? {}),
+    ...(win.AMP_EXP ?? {}),
```

Comment:
Due to the `??=` above, we've can assert that `AMP_EXP` is defined on `win`.

```suggestion
    ...win.AMP_EXP,
```

Reply:
Oops didn't notice it was a commit suggestion... pushed it in a separate commit